runtime/lua/vim/lsp/protocol.lua

```diff
@@ -835,7 +835,10 @@ function protocol.make_client_capabilities()
         refreshSupport = true,
       },
       didChangeWatchedFiles = {
-        dynamicRegistration = true,
+        -- TODO(lewis6991): do not advertise didChangeWatchedFiles on Linux
+        -- since all the current backends are too limited.
+        -- Ref: #27807, #28058, #23291, #26520
+        dynamicRegistration = vim.uv.os_uname().sysname ~= 'Linux',
```

Comment:
Can you add openbsd too? That's even more limited

Reply:
Sure but I don't know the exact string to match. Maybe I should just include Mac and Windows instead.